tests: Bluetooth: tester: Add HFP HF support

jerryyang35 opened the pull request:
Add Bluetooth Tester Protocol (BTP) support for Hands-Free Profile in Hands-Free (HF) role.

BTP command handlers for HFP HF operations including connection management, call control, audio connection, volume control, Codec negotiation, voice recognition, and various HFP features.

Event notifications for connection state changes, call state changes, SCO audio connection events, and feature-specific callback.

Supported HFP HF features including,
- Caller Line Identification (CLI)
- Volume control (VGM/VGS)
- Codec negotiation (CVSD/mSBC)
- Echo Cancellation and Noise Reduction (ECNR)
- Three-way calling
- Enhanced Call Control (ECC)
- Voice recognition (basic and enhanced)
- Voice tag
- HF indicators


The change for auto-pts is [here](https://github.com/auto-pts/auto-pts/pull/1655).

ChangNice commented on tests/bluetooth/tester/src/btp/btp_hfp.h:
```diff
@@ -0,0 +1,361 @@
+} __packed;
+
+struct btp_hfp_disable_slc_rp {
+	
```
add a member "status" may be a good practice.

ChangNice commented on tests/bluetooth/tester/src/btp/btp_hfp.h:
```diff
@@ -0,0 +1,361 @@
+	HFP_OUT_CALL                  = 0x06,
+	HFP_CLS_MEM_CALL_LIST         = 0x07,
+	HFP_OUT_MEM_CALL              = 0x08,
+	HFP_OUT_MEM_OUTOFRANGE_CALL   = 0x9,
```
format not align with previous number 0x08

jerryyang35 commented on tests/bluetooth/tester/src/btp/btp.h:
```diff
@@ -81,8 +82,9 @@
 #define BTP_SERVICE_ID_TMAP     0x1c
 #define BTP_SERVICE_ID_OTS      0x1d
 #define BTP_SERVICE_ID_PBP      0x1e
+#define BTP_SERVICE_ID_HFP      0x21
```
0x1f is for rfcomm in another PR

makeshi commented on tests/bluetooth/tester/src/btp_hfp_ag.c:
```diff
@@ -0,0 +1,2275 @@
+
+static struct hfp_ag_connection *find_connection_by_call(struct bt_hfp_ag_call *call)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(ag_connections); i++) {
```
Do we need a lock here for some protection？

sjanc reviewed (commented):
btw it would help if this is at least split into 2 PRs (one for AG and one for HF)

lylezhu2012 committed:
tests: Bluetooth: tester: Add HFP HF support

Add Bluetooth Tester Protocol (BTP) support for Hands-Free Profile in
Hands-Free (HF) role.

BTP command handlers for HFP HF operations including connection
management, call control, audio connection, volume control, Codec
negotiation, voice recognition, and various HFP features.

Event notifications for connection state changes, call state changes,
SCO audio connection events, and feature-specific callback.

Supported HFP HF features including,
- Caller Line Identification (CLI)
- Volume control (VGM/VGS)
- Codec negotiation (CVSD/mSBC)
- Echo Cancellation and Noise Reduction (ECNR)
- Three-way calling
- Enhanced Call Control (ECC)
- Voice recognition (basic and enhanced)
- Voice tag
- HF indicators

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

lylezhu2012 commented:
> btw it would help if this is at least split into 2 PRs (one for AG and one for HF)

The PR for HFP AG is [here](https://github.com/zephyrproject-rtos/zephyr/pull/108632).
